doc(pe): Document policy options (#601)
armory/docs: content/en/docs/plugin-guide/plugin-policy-engine/packages/spinnaker.http.authz/tasks/type.deleteManifest.md

---
title: "Task Type: deleteManifest"
linktitle: "deleteManifest"

> description: "Policy controls whether or not a deleteManifest that is triggered from outside a spinnaker pipeline (e.g. from the 'Clusters' tab of an application) can run."

Reviewer: ```suggestion
description: "Policy that controls whether or not a deleteManifest that is triggered from outside of a pipeline (such as from the **Clusters** tab of an application) can run."
```

---

- **Path:** tasks
- **Method:** Post
- **Package:** `spinnaker.http.authz`

## Compatibility Note

Starting in 2.26, the UI has been updated to more closely follow immutable infrastructure principles.

When you navigate to the Infrastructure tab in the UI for an application that has the Kubernetes provider configured, actions that change the Kubernetes infrastructure (such as Create or Delete), including Clusters, Load Balancers, and Firewalls, are no longer available.

### Impact

Users do not see these actions in the UI by default. You must configure the UI to display them if you want your users to be able to perform them through the UI.

### Workaround

Whether or not these actions are available in the UI is controlled by the following property in `settings-local.yml`:

```yml
window.spinnakerSettings.kubernetesAdHocInfraWritesEnabled = <boolean>;
```

> This setting must be applied in order for policy to conditionally allow this functionality.

Reviewer: ```suggestion
This setting must be applied in order for this type of policy to conditionally allow this functionality.
```

Reviewer: I think this sentence appears in a few places, and I missed this typo.

Can also replace `policy` with `Policy Engine`, which I guess is more accurate.

## Example Payload

<details><summary>Click to expand</summary>

```json
{
  "input": {
    "body": {
      "application": "hostname",
      "description": "Delete manifest",
      "job": [
        {
          "account": "spinnaker",
          "cloudProvider": "kubernetes",
          "location": "staging",
          "manifestName": "deployment hostname",
          "options": {
            "gracePeriodSeconds": 5,
            "orphanDependants": false
          },
          "reason": null,
          "type": "deleteManifest",
          "user": "myUserName"
        }
      ]
    },
    "method": "POST",
    "path": [
      "tasks"
    ],
    "user": {
      "isAdmin": false,
      "roles": [],
      "username": "myUserName"
    }
  }
}
```
</details>

## Example Policy

> This example will prevent users from deleting deployed manifests from production accounts on the 'Clusters' tab of the spinnaker UI.

Reviewer: ```suggestion
This example will prevent users from deleting deployed manifests from production accounts on the **Clusters** tab of the UI.
```

isProductionAccount(account){
	["prodAccount1","prodAccount2"][_]==account
}

message = "Manifests cannot be deleted outside of a pipeline in production accounts"{
      createsTaskOfType("deleteManifest")
      isProductionAccount(input.body.job[_].account)
}
createsTaskOfType(tasktype){
    input.method="POST"
    input.path=["tasks"]
    input.body.job[_].type=tasktype
}
{{< /prism >}}

## Keys

| Key                                           | Type      | Description                                                                                                                        |
| :-------------------------------------------- | --------- | ---------------------------------------------------------------------------------------------------------------------------------- |
| `input.body.application`                      | `string`  | The name of the application that deployed the manifest being deleted.                                                              |
| `input.body.description`                      | `string`  | The phrase "Delete Manifest".                                                                                                      |

> | `input.body.job[].account`                    | `string`  | The spinnaker account that will delete the manifest.                                                                               |

Reviewer: ```suggestion
| `input.body.job[].account`                    | `string`  | The Armory Enterprise account that will delete the manifest.                                                                               |
```
